streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamPrintTest.java

```diff
@@ -41,84 +42,83 @@
 
 public class KStreamPrintTest {
 
-    private final String topicName = "topic";
     private final Serde<Integer> intSerd = Serdes.Integer();
     private final Serde<String> stringSerd = Serdes.String();
     private PrintWriter printWriter;
     private ByteArrayOutputStream byteOutStream;
     @Rule
     public KStreamTestDriver driver = new KStreamTestDriver();
 
+    private KeyValueMapper<Integer, String, String> mapper;
+    private KStreamPrint kStreamPrint;
+
     @Before
     public void setUp() {
         byteOutStream = new ByteArrayOutputStream();
         printWriter = new PrintWriter(new OutputStreamWriter(byteOutStream, StandardCharsets.UTF_8));
-    }
 
-    @Test
-    public void testPrintKeyValueWithName() {
-        KeyValueMapper<Integer, String, String> mapper = new KeyValueMapper<Integer, String, String>() {
+        mapper = new KeyValueMapper<Integer, String, String>() {
             @Override
-            public String apply(Integer key, String value) {
+            public String apply(final Integer key, final String value) {
                 return String.format("%d, %s", key, value);
             }
         };
-        final KStreamPrint<Integer, String> kStreamPrint = new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, "test-stream"), intSerd, stringSerd);
+
+        kStreamPrint = new KStreamPrint<>(new PrintForeachAction<>(printWriter, mapper, "test-stream"), intSerd, stringSerd);
+    }
+
+    @Test
+    public void testPrintStreamWithProvidedKeyValueMapper() {
 
         final List<KeyValue<Integer, String>> inputRecords = Arrays.asList(
                 new KeyValue<>(0, "zero"),
                 new KeyValue<>(1, "one"),
                 new KeyValue<>(2, "two"),
                 new KeyValue<>(3, "three"));
-        
+
         final String[] expectedResult = {"[test-stream]: 0, zero", "[test-stream]: 1, one", "[test-stream]: 2, two", "[test-stream]: 3, three"};
-        
-        final StreamsBuilder builder = new StreamsBuilder();
-        final KStream<Integer, String> stream = builder.stream(intSerd, stringSerd, topicName);
-        stream.process(kStreamPrint);
-        
-        driver.setUp(builder);
+
+        Processor printProcessor = kStreamPrint.get();
```

Comment:
You could move this block from L81 -> L85 into setup. Then you can remove it from here and the next method. Also, add SuppressWarnings to both tests.